src/CMakeLists.txt

```diff
@@ -70,6 +67,9 @@ if(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
   endif()
 endif(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
 
+set(CMAKE_ASM_COMPILER  ${PROJECT_SOURCE_DIR}/src/yasm-wrapper)
+set(CMAKE_ASM_FLAGS "-f elf64")
+
```

Comment:
could you explain a little bit the rationale of the changes in your commit message?